src/FixerConfiguration/FixerOption.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+final class FixerOption implements FixerOptionInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $description;
+
+    /**
+     * @var mixed
+     */
+    private $default;
+
+    /**
+     * @var bool
+     */
+    private $isRequired = true;
+
+    /**
+     * @var null|string[]
+     */
+    private $allowedTypes;
+
+    /**
+     * @var null|array
+     */
+    private $allowedValues;
+
+    /**
+     * @var \Closure|null
+     */
+    private $normalizer;
+
+    /**
+     * @param string        $name
+     * @param string        $description
+     * @param bool          $isRequired
+     * @param mixed         $default
+     * @param string[]|null $allowedTypes
+     * @param array|null    $allowedValues
+     * @param \Closure|null $normalizer
+     */
+    public function __construct(
+        $name,
+        $description,
+        $isRequired = true,
+        $default = null,
+        array $allowedTypes = null,
+        array $allowedValues = null,
+        \Closure $normalizer = null
+    ) {
+        if (null !== $allowedValues) {
+            foreach ($allowedValues as &$allowedValue) {
+                if ($allowedValue instanceof \Closure) {
+                    $allowedValue = $this->unbind($allowedValue);
+                }
+            }
+        }
+
+        $this->name = $name;
+        $this->description = $description;
+        $this->isRequired = $isRequired;
+        $this->default = $default;
```

Comment:
for some reason GitHub shows it outdated, but it's not:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2500#discussion_r106368247